niceman/resource/base.py

```diff
@@ -182,11 +182,11 @@ def _discover_types():
     def _find_resources(self, resref, resref_type):
         def from_name(x):
             return [(name, config) for name, config in self.inventory.items()
-                    if x == name]
+                    if name.startswith(x)]
```

Comment:
Hmm, I'm not sure that we should bother supporting partial name matches.  A name should already serve as a short label, whereas partial matching of an ID is useful because they can be quite long and an initial substring likely to be unique.  I'm basically viewing this as git's hexsha versus a refname.

In addition to matching my expectations, doing so should reduce the complexity of this PR.  With `from_id`, it'd be the current pass to see if there are full matches, and, if none, a second `startswith` pass, I think.

Reply:
Agree about names! It did feel weird and too magic (although my ssh aliases all short) to go by  names abbreviations. Oh well, at least I had fun. I also at the end felt that just having to passes might be cleaner. Will see...